index.bs

```diff
@@ -1462,3 +1485,278 @@ The [=remote end steps=] with |command parameters| are:
+
+### Definition ### {#module-browsingContext-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+BrowsingContextCommand = (BrowsingContextGetTreeCommand)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+BrowsingContextResult = (BrowsingContextGetTreeResult)
+
+BrowsingContextEvent = (
+    BrowsingContextCreatedEvent //
+    BrowsingContextDestroyedEvent
+)
+
+</pre>
+
+### Types ### {#module-browsingcontext-types}
+
+#### The browsingContext.BrowsingContext Type #### {#types-browsingcontext}
+
+[=remote end definition=] and [=local end definition=]
+
+<pre class="cddl remote-cddl local-cddl">
+BrowsingContext = text;
+</pre>
+
+Each [=/browsing context=] has an associated <dfn export>browsing context
+id</dfn>, which is a string uniquely identifying that browsing context. This is
+implicitly set when the context is created. For browsing contexts with an
+associated WebDriver [=window handle=] the [=/browsing context id=] must be the
+same as the [=window handle=].
+
+<div algorithm>
+To <dfn>get a browsing context</dfn> given |context id|:
+
+ 1. If |context id| is null, return [=success=] with data null.
+
+ 1. If there is no browsing context with [=browsing context id=] |context id| return
+    [=error=] with [=error code=] [=no such frame=]
+
+ 1. Let |context| be the browsing context with id |context id|.
+
+ 1. Return [=success=] with data |context|
+
+</div>
+
+#### The browsingContext.BrowsingContextInfo Type #### {#types-browsingcontextinfo}
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+BrowsingContextInfo = {
+  context: BrowsingContext,
+  parent: BrowsingContext / null,
+  url: text,
+  children: [*BrowsingContextInfo] / null
+}
+
+</pre>
+
+The <code>BrowsingContextInfo</code> type represents the properties of a
+browsing context.
+
+<div algorithm> To <dfn>get the browsing context info</dfn> given |context|,
+|depth| and |max depth|:
+
+ 1. Let |context id| be the [=browsing context id=] for |context|.
+
+ 1. If |context| has a [=parent browsing context=] let |parent id| be the
+    [=browsing context id=] of that parent. Otherwise let |parent id| be null.
+
+ 1. Let |document| be |context|'s [=active document=].
+
+ 1. Let |url| be the result of running the [=URL serializer=], given
+    |document|'s <a spec=dom>URL</a>.
+
+    Note: This includes the fragment component of the URL.
+
+ 1. Let |child info| be the result of [=get the ancestor browsing contexts=]
+    given |context id|, |depth| + 1, and |max depth|.
+
+ 1. Let |context info| be a [=map=] matching the <code>BrowsingContextInfo</code>
+    production with the <code>context</code> field set to |context id|, the
+    <code>parent</code> field set to |parent id|, the <code>url</code> field
+    set to |url|, and the <code>children</code> field set to |child info|.
+
+ 1. Return |context info|.
+
+</div>
+
+<div algorithm>
+  To <dfn>get the ancestor browsing contexts</dfn> given |parent id|, |depth|
```

Comment:
Why is it called get the ancestor browsing contexts instead of get the children/descendant browsing contexts?
From step 5, it seems to be collecting the child contexts.